src/World.cpp

```diff
@@ -1059,12 +1060,23 @@ void cWorld::Tick(std::chrono::milliseconds a_Dt, std::chrono::milliseconds a_La
 
 	if (m_WorldAge - m_LastSave > std::chrono::minutes(5))  // Save each 5 minutes
 	{
+		LOG("######## PERIODIC SAVE");
 		SaveAllChunks();
 	}
 
+
 	if (m_WorldAge - m_LastUnload > std::chrono::seconds(10))  // Unload every 10 seconds
 	{
+		LOG("######## %d / %d", GetNumUnusedDiryChunks(), m_UnusedDirtyChunksCap);
 		UnloadUnusedChunks();
+		if (m_WorldAge - m_LastSave > std::chrono::seconds(30))
```

Comment:
The outer 10 sec block does not guarantee 10 sec spacing between all saves.

If a periodic save happens right before the 10 second block is entered, another save would immediately follow.

I guess this is a bit confusing because I hijacked the `m_LastUnload` variable for saving.